### PR TITLE
Allow customizing the title of each app served with pn.serve

### LINKED ISSUE #1292: Improve Title functionality of panel.serve to be specific for each app served
THE BELOW IS COPIED FROM https://discourse.holoviz.org/t/how-do-i-set-the-app-title-with-panel-serve/550

IN THAT POST I HAVE TRIED DIFFERENT APPROACHES WITHOUT LUCK.

#### SOLUTION

Provide functionality to support use case

-----------

I'm using `panel.serve` to serve a list of Panel applications.

I would like the *index* page to show a custom title in the browser tab like `Corp Analytics`

I would like each app to show a custom title in the browser tab like `App Name`.

**How do I achieve that?**

What I can do and would like is shown below

#### Index Page

image

#### App 1 Page

image

#### App 2 Page

image

#### Code

```python
import panel as pn

def create_app():

    return pn.Spacer(background="green", sizing_mode="stretch_both")

APP_ROUTES = {

    "app1": create_app,

    "app2": create_app

}

pn.serve(APP_ROUTES, port=14033, title="My App")
```

### PR DESCRIPTION
**Pull request by maximlt**
Fixes #1292 

@philippjfr When you have the time please let me know if you're fine with the tests :)

### CODE REVIEW

File: examples/user_guide/Deploy_and_Export.ipynb [hunk 1]
```diff
@@ -229,6 +229,14 @@
     "pn.serve({'markdown': '# This is a Panel app', 'json': pn.pane.JSON({'abc': 123})})\n",
     "```\n",
     "\n",
+    "You can customize the HTML title of each application by supplying a dictionnary where the keys represent the URL slugs and the values represent the titles, e.g.:\n",
```
Comment (philippjfr) [inline examples/user_guide/Deploy_and_Export.ipynb:232]:
```suggestion
    "You can customize the HTML title of each application by supplying a dictionary where the keys represent the URL slugs and the values represent the titles, e.g.:\n",
```

File: panel/io/server.py [hunk 2]
```diff
@@ -130,8 +130,9 @@ def serve(panels, port=0, websocket_origin=None, loop=None, show=True,
       Whether to open the server in a new browser tab on start
     start : boolean(optional, default=False)
       Whether to start the Server
-    title: str (optional, default=None)
-      An HTML title for the application
+    title: str or {str: str} (optional, default=None)
+      An HTML title for the application or a dictionnary mapping
```
Comment (philippjfr) [inline panel/io/server.py:134]:
```suggestion
      An HTML title for the application or a dictionary mapping
```

File: panel/io/server.py [hunk 3]
```diff
@@ -188,8 +189,9 @@ def get_server(panel, port=0, websocket_origin=None, loop=None,
...
```
Comment (philippjfr) [inline panel/io/server.py:193]:
```suggestion
      An HTML title for the application or a dictionary mapping
```

File: panel/io/server.py [hunk 4]
```diff
@@ -210,6 +212,16 @@ def get_server(panel, port=0, websocket_origin=None, loop=None,
     if isinstance(panel, dict):
         apps = {}
         for slug, app in panel.items():
+            if isinstance(title, dict):
+                try:
+                    title_ = title[slug]
+                except KeyError:
+                    raise KeyError(
+                        "Keys of the title dictionnary and of the apps "
+                        f"dictionnary must match. No {slug} key found in the "
```
Comment (philippjfr) [inline panel/io/server.py:221]:
```suggestion
                        f"dictionary must match. No {slug} key found in the "
```

At some point we should convert to f-strings throughout.

**Comment by maximlt** [comment]
Oops got confused with the French word... Thanks for the edits, I guess that you can merge it now, or do I still have something to do?

**Pull request merged**